clients/stellar-relay-lib/src/config.rs

```diff
@@ -0,0 +1,277 @@
+
+		let _: ConnectionInfoCfg =
+			serde_json::from_str(json).expect("should return a ConnectionInfoCfg");
+	}
+
+	#[test]
+	fn missing_fields_in_connection_info_config() {
+		// missing port
+		let json = r#"
+		 {
+             "address":"1.2.3.4",
+             "secret_key": "SBLI7RKEJAEFGLZUBSCOFJHQBPFYIIPLBCKN7WVCWT4NEG2UJEW33N73",
+             "recv_scp_msgs":true,
+             "remote_called_us":false
+		 }
+		 "#;
+
+		assert!(serde_json::from_str::<ConnectionInfoCfg>(json).is_err());
+
+		// missing secret key
+		let json = r#"
+		 {
+             "address":"1.2.3.4",
+             "port":11625,
+             "auth_cert_expiration":0,
+             "recv_tx_msgs":false,
+             "recv_scp_msgs":true
+		 }
+		 "#;
+		assert!(serde_json::from_str::<ConnectionInfoCfg>(json).is_err());
+	}
+
+	#[test]
+	fn node_info_conversion_successful() {
+		let json = r#"
+        {
+			"ledger_version":19,
+			"overlay_version":25,
+			"overlay_min_version":23,
+			"version_str":"v19.5.0",
+			"is_pub_net": false
+		}
+        "#;
+
+		let _: NodeInfoCfg = serde_json::from_str(json).expect("should return a NodeInfoCfg");
+	}
+
+	#[test]
+	fn missing_fields_in_node_info_config() {
+		// missing version_str
+		let json = r#"
+        {
+			"ledger_version":19,
+			"overlay_version":25,
+			"overlay_min_version":23,
+			"is_pub_net": false
+		}
+        "#;
+
+		assert!(serde_json::from_str::<NodeInfoCfg>(json).is_err());
+
+		// missing is_pub_net
+		let json = r#"
+        {
+			"ledger_version":19,
+			"overlay_version":25,
+			"overlay_min_version":23,
+			"version_str":"v19.5.0"
+		}
+        "#;
+
+		assert!(serde_json::from_str::<NodeInfoCfg>(json).is_err());
+	}
+
+	#[test]
+	fn stellar_relay_config_conversion_successful() {
+		let json = r#"
+        {
+            "connection_info":{
+                "address":"1.2.3.4",
+                "port":11625,
+                "secret_key": "SBLI7RKEJAEFGLZUBSCOFJHQBPFYIIPLBCKN7WVCWT4NEG2UJEW33N73",
+                "auth_cert_expiration":0,
+                "recv_scp_msgs":true
+            },
+            "node_info":{
+                "ledger_version":19,
+                "overlay_version":25,
+                "overlay_min_version":23,
+                "version_str":"v19.5.0",
+                "is_pub_net":false
+            }
+        }
+		"#;
+
+		let _: StellarOverlayConfig =
+			serde_json::from_str(json).expect("should return a NodeInfoCfg");
+	}
+
+	#[test]
+	fn missing_fields_in_stellar_relay_config() {
+		// missing address in Connection_Info, and overlay_min_version in Node_Info
+		let json = r#"
+        {
+            "connection_info":{
+                "port":11625,
+                "secret_key": "SBLI7RKEJAEFGLZUBSCOFJHQBPFYIIPLBCKN7WVCWT4NEG2UJEW33N73",
+            },
+            "node_info":{
+                "ledger_version":19,
+                "overlay_version":25,
+                "version_str":"v19.5.0",
+                "is_pub_net":false
+            }
+        }
```

Comment:
Can we maybe indent all these json strings properly? I know it's annoying because we have to do it manually since the linter won't do it automatically to prevent 'destroying' the struct but it looks a bit weird.